src/code/sys_matrix.c

```diff
@@ -470,59 +470,64 @@ void Matrix_TranslateRotateZYX(Vec3f* translation, Vec3s* rotation) {
     }
 }
 
-void func_800D1694(f32 x, f32 y, f32 z, Vec3s* vec) {
+/**
+ * Translate and rotate using YXZ Tait-Bryan angles.
+ * This means a (column) vector is first rotated around Z, then around X, then around Y, then translated, then gets
+ * transformed according to whatever the matrix was previously.
+ */
+void Matrix_TranslateRotateYXZ(f32 translateX, f32 translateY, f32 translateZ, Vec3s* rot) {
```

Comment:
This one feels wrong to me because the function isn't quite "equivalent" to `Matrix_TranslateRotateZYX` implementation-wise. More specifically, the previous matrix seems to only be partially preserved, and only rotation Z carries over? It seems like that to me when comparing the implementaton of each section, but I could be wrong.
If I'm right, then I think at least the comment needs to be adjusted, and we might want a more specific name, because with the current name this function sounds like another version of `Matrix_TranslateRotateZYX` with a different rotation order, which could be misleading.